libs/langchain/langchain/vectorstores/neo4j_vector.py

```diff
@@ -0,0 +1,626 @@
+from __future__ import annotations
+
+import enum
+import logging
+import uuid
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+)
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.vectorstores.base import VectorStore
+
+
+class DistanceStrategy(str, enum.Enum):
+    """Enumerator of the Distance strategies."""
+
+    EUCLIDEAN = "euclidean"
+    COSINE = "cosine"
+
+
+DEFAULT_DISTANCE_STRATEGY = DistanceStrategy.COSINE
+
+
+class Neo4jVector(VectorStore):
+    """`Neo4j` vector index.
+
+    To use, you should have the ``neo4j`` python package installed.
+
+    Args:
+        url: Neo4j connection url
+        username: Neo4j username.
+        password: Neo4j password
+        database: Optionally provide Neo4j database
+                  Defaults to "neo4j"
+        embedding_function: Any embedding function implementing
+            `langchain.embeddings.base.Embeddings` interface.
+        distance_strategy: The distance strategy to use. (default: COSINE)
+        pre_delete_collection: If True, will delete existing data if it exists.
+            (default: False). Useful for testing.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.vectorstores.neo4j_vector import Neo4jVector
+            from langchain.embeddings.openai import OpenAIEmbeddings
+
+            url="bolt://localhost:7687"
+            username="neo4j"
+            password="pleaseletmein"
+            embeddings = OpenAIEmbeddings()
+            vectorestore = Neo4jVector.from_documents(
+                embedding=embeddings,
+                documents=docs,
+                url=url
+                username=username,
+                password=password,
+            )
+
+
+    """
+
+    def __init__(
+        self,
+        username: str,
+        password: str,
```

Comment:
up to you, but most integrations allow credentials to also be read from environmental vars (so you'd make this optional here and have something like a `NEO4V_PASSWORD` env var you check for if this is None